mypy/build.py

```diff
@@ -861,19 +865,68 @@ def is_file(path: str) -> bool:
     return res
 
 
-def find_module(id: str, lib_path_arg: Iterable[str]) -> Optional[str]:
+USER_SITE_PACKAGES = \
+    '"import site;print(site.getusersitepackages());print(*site.getsitepackages(), sep=\'\\n\')"'
+VIRTUALENV_SITE_PACKAGES = \
+    '"from distutils.sysconfig import get_python_lib;print(get_python_lib())"'
+
+
+def call_python(python: str, command: str) -> str:
+    return check_output([python, '-c', command]).decode(sys.stdout.encoding)
+
+
+def get_package_dirs(python: Optional[str]) -> List[str]:
+    """Find package directories for given python (default to Python running
+    mypy)."""
+    global package_dirs_cache
+    if package_dirs_cache:
+        return package_dirs_cache
+    package_dirs = []  # type: List[str]
+    if python:
+        # Use subprocess to get the package directory of given Python
+        # executable
+        check = check_output([python, '-V'], stderr=STDOUT).decode('UTF-8')
+        if not check.startswith('Python'):
+            return package_dirs
+        # If we have a working python executable, query information from it
+        output = call_python(python, USER_SITE_PACKAGES)
+        for line in output.splitlines():
+            if os.path.isdir(line):
+                package_dirs.append(line)
+        if not package_dirs:
+            # if no paths are found, we fall back on sysconfig
+            output = call_python(python, VIRTUALENV_SITE_PACKAGES)
+            for line in output.splitlines():
+                if os.path.isdir(line):
+                    package_dirs.append(line)
+    else:
+        # Use running Python's package dirs
+        try:
+            user_dir = site.getusersitepackages()
+            package_dirs = site.getsitepackages() + [user_dir]
+        except AttributeError:
+            package_dirs = [get_python_lib()]
+    package_dirs_cache = package_dirs
```

Comment:
This global assignment is a bit jarring. Maybe drop the `global` and use `package_dirs_cache[:] = package_dirs` here?